### PR TITLE
Flappy Bird Overhaul

### LINKED ISSUE #2: Collisions
When the bird collides with an obstacle, check if the tag is equal to pipe. If so move to a game over screen.

*Recommended to change scenes or activate an sprite and/or activate a canvas within the scene

### LINKED ISSUE #3: Moving Objects
Spawn pipes that move across the screen. Once the pipes are not seen by the player delete the game object. Apply rectangle collider on the pipe and set to trigger.

*Recommended to use Rigidbody or edit transform.position. Make sure the pipes move relative to the game speed than the frame rate.

### PR DESCRIPTION
**Pull request by Galaxy25**
Closes #3 
Closes #2 
A general overhaul to flappy bird. Creates movable pipes that repeatably spawn. Movement matches the original flappy bird. The bird dies on collision.

### CODE REVIEW

**Comment by Galaxy25** [comment]
Approved

**Comment by BrandonPacewic** [comment]
@Galaxy25 Looks like there are some merge conflicts with `Assets/Scenes/SampleScene.unity`

**Comment by Galaxy25** [comment]
@BrandonPacewic Forgot to rename the scene so GitHub wouldn't try to merge them together. It should now work. Also I ran clang.

File: Assets/Scripts/Movement.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
+using System.Collections;
```
Comment (BrandonPacewic) [inline Assets/Scripts/Movement.cs:4]: Need to add our copyright

File: Assets/Scripts/PipeMovement.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
+using System.Collections;
```
Comment (BrandonPacewic) [inline Assets/Scripts/PipeMovement.cs:4]: Copyright.

File: Assets/Scripts/Spawn.cs [hunk 3]
```diff
@@ -0,0 +1,29 @@
+using System.Collections;
```
Comment (BrandonPacewic) [inline Assets/Scripts/Spawn.cs:4]: Copyright

File: Assets/Scripts/PipeMovement.cs [hunk 4]
```diff
@@ -0,0 +1,17 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class PipeMovement : MonoBehaviour
+{
+    public float speed = 5f;
+
+    void Update()
```
Comment (BrandonPacewic) [inline Assets/Scripts/PipeMovement.cs]:
```suggestion
    private void Update()
```

File: Assets/Scripts/Spawn.cs [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Spawn : MonoBehaviour
+{
+    public float endTimer = 5f;
+    private float timer;
+    public GameObject pipe;
+
+    void Start()
```
Comment (BrandonPacewic) [inline Assets/Scripts/Spawn.cs]:
```suggestion
    private void Start()
```

File: Assets/Scripts/Spawn.cs [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    {
+        timer = endTimer;
+    }
+
+    // Update is called once per frame
+    void Update()
```
Comment (BrandonPacewic) [inline Assets/Scripts/Spawn.cs]:
```suggestion
    private void Update()
```

File: Assets/Scripts/Movement.cs [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Movement : MonoBehaviour
+{
+    public Rigidbody2D rb;
+    public float force = 800f;
+    public bool canJump = true;
+
+    void Update()
```
Comment (BrandonPacewic) [inline Assets/Scripts/Movement.cs]:
```suggestion
    private void Update()
```

File: Assets/Scripts/Movement.cs [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+            rb.velocity = new Vector3(0, 0, 0);
+            rb.AddForce(transform.up * force, ForceMode2D.Impulse);
+            canJump = false;
+        }
+        else if (Input.GetButtonUp("Jump"))
+        {
+            canJump = true;
+        }
+    }
+    void OnTriggerEnter2D(Collider2D trigger)
```
Comment (BrandonPacewic) [inline Assets/Scripts/Movement.cs]:
```suggestion
    }
   
    private void OnTriggerEnter2D(Collider2D trigger)
```

**Review by BrandonPacewic — CHANGES_REQUESTED** [review]
General nitpicks. I'll push a fix in a few minutes.

**Comment by BrandonPacewic** [comment]
Also updated to Unity version `2021.3.12f1`.

File: Assets/Scripts/Movement.cs [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Movement : MonoBehaviour
+{
+    public Rigidbody2D rb;
+    public float force = 800f;
+    public bool canJump = true;
```
Comment (BrandonPacewic) [inline Assets/Scripts/Movement.cs:12]: Do these need to be public?

File: Assets/Scripts/PipeMovement.cs [hunk 10]
```diff
@@ -0,0 +1,20 @@
+// Copyright (c) TigardHighGDC
+// SPDX-License SPDX-License-Identifier: Apache-2.0
+
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class PipeMovement : MonoBehaviour
+{
+    public float speed = 5f;
```
Comment (BrandonPacewic) [inline Assets/Scripts/PipeMovement.cs:10]: Does this need to be public?

File: Assets/Scripts/Spawn.cs [hunk 11]
```diff
@@ -0,0 +1,31 @@
+// Copyright (c) TigardHighGDC
+// SPDX-License SPDX-License-Identifier: Apache-2.0
+
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Spawn : MonoBehaviour
+{
+    public float endTimer = 5f;
```
Comment (BrandonPacewic) [inline Assets/Scripts/Spawn.cs:10]: Does this need to be public?

**Review by BrandonPacewic — CHANGES_REQUESTED** [review]
If we have a variable public then shouldn't it be set from the object settings rather than a fixed value?

**Comment by Galaxy25** [comment]
Having a public variable makes it easier to edit within Unity allowing it to be tweaked faster. Since the game was a test and not a multiple-month project rechanging the variables into private would cause more problems than preventing future ones.

**Comment by Galaxy25** [comment]
Also another use case for public variables would be allowing different types of information upon the game object creation. For example enemies with varying movement speed

**Review by BrandonPacewic — DISMISSED** [review]

**Comment by BrandonPacewic** [comment]
I'm getting ready to merge and updating the change log - please @me if any other changes are pushed.

**Comment by BrandonPacewic** [comment]
@Galaxy25 Before I merge, is it correct to leave `Flying.cs` in addition to `Movement.cs`?

**Comment by Galaxy25** [comment]
Flying.cs needs to be removed.
Then Approved

**Review by BrandonPacewic — APPROVED** [review]
In the future I will want to discuss file naming. For instance:

- `Movement.cs` -> `BirdMovementController.cs`
- `PipeMovement.cs` -> `PipeMovementController.cs`
- `Spawn.cs` -> `PipeSpawner.cs`

But like you said, this is supposed to be janky for now 🙃

**Comment by BrandonPacewic** [comment]
Thanks for getting this done! Next time we should get @JakeSmig to help with the review process 🖥️

**Pull request merged**